api/datasets.go

```diff
@@ -197,14 +199,72 @@ func extensionToMimeType(ext string) string {
 	}
 }
 
-func (h *DatasetHandlers) listHandler(w http.ResponseWriter, r *http.Request) {
-	args := lib.ListParamsFromRequest(r)
-	args.OrderBy = "created"
+// snoop reads from an io.ReadCloser and restores it so it can be read again
```

Comment:
This should probably live elsewhere.

Reply:
I'd generally like to avoid needing this snoop method, but if it needs to exist for now to keep things moving, then so be it

Reply:
For now it's required but agree it should go away. It's a code smell we have to carry around until the refactor of the endpoints is done.